### PR TITLE
feat(google-calendar): add check if day already has event

### PR DESCRIPTION
**Pull request by PatrickDohn**
#### Summary

_Enumerate and describe your changes_

- add a check to see if a sick or vacation day has already been taken on the day requested

#### Checklist

_Double check list and complete these actions before merge_

- [x] PR title is in format `TYPE(scope): description`
- [x] Link PR to Issue (if any issues are related)

### CODE REVIEW

File: src/utils/googleCalendarUtil.js [hunk 1]
```diff
@@ -18,10 +18,25 @@ class GoogleCalendarUtil {
     });
   }
 
+  // add check for for if this function was called today
   async createVacationEvent(slackName) {
     await this.createTodayEvent(slackName, 'vacation');
   }
 
+  async getAllEvents(startTime, endTime) {
```
Comment (davidholyko) [inline src/utils/googleCalendarUtil.js:26]: `getAllSingleDayEvents`

File: src/app.js [hunk 2]
```diff
@@ -88,10 +89,23 @@ const onResponse = async (response) => {
     // into smaller files
     case SET_SICK: {
       try {
-        await GoogleSheetUtil.setTodayAsSickDay(slackName);
-        await GoogleCalendarUtil.createSickEvent(slackName);
-        // TODO: add fun emoji
-        say(`Hello <@${event.user}>! Sick day marked. Hope you feel better!`);
+        const items = await GoogleCalendarUtil.getAllEvents(
+          moment().startOf('day').format(),
+          moment().endOf('day').format(),
```
Comment (davidholyko) [inline src/app.js:94]: this logic should live inside of the util function itself

Comment (PatrickDohn) [reply src/app.js:94]: The whole function call? or the momentjs stuff? The moments are the required arguments for the function.

File: src/utils/googleCalendarUtil.js (hunk 1 above)
Comment (PatrickDohn) [reply src/utils/googleCalendarUtil.js:26]: Fixed

File: src/app.js (hunk 2 above)
Comment (davidholyko) [reply src/app.js:94]: moment stuff. it can be a part of the function and not an argument

File: src/app.js [hunk 3]
```diff
@@ -88,10 +89,23 @@ const onResponse = async (response) => {
...
+        );
+        let eventList = [];
```
Comment (davidholyko) [inline src/app.js:96]: its kind of strange your linter is not picking up these let assignments. these should be const's

File: src/app.js [hunk 4]
```diff
@@ -88,10 +88,20 @@ const onResponse = async (response) => {
...
-        await GoogleSheetUtil.setTodayAsSickDay(slackName);
-        await GoogleCalendarUtil.createSickEvent(slackName);
-        // TODO: add fun emoji
-        say(`Hello <@${event.user}>! Sick day marked. Hope you feel better!`);
+        const items = await GoogleCalendarUtil.getAllSingleDayEvents();
+        let eventList = [];
+        const summaryCheck = `${slackName}: Sick OOO`;
+        for (let i = 0; i < items.length; i++) {
+          eventList.push(items[i].summary);
+        }
```
Comment (davidholyko) [inline src/app.js:96]: user lodash `map`

File: src/app.js [hunk 5]
```diff
@@ -88,10 +88,20 @@ const onResponse = async (response) => {
...
+        if (eventList.includes(summaryCheck)) {
```
Comment (davidholyko) [inline src/app.js:97]: lodash `includes`

File: src/app.js [hunk 6]
```diff
@@ -88,10 +88,20 @@ const onResponse = async (response) => {
...
+          say(`Hello <@${event.user}>! Today was already marked as a sick day.`);
+        } else {
+          await GoogleSheetUtil.setTodayAsSickDay(slackName);
+          await GoogleCalendarUtil.createSickEvent(slackName);
+          // TODO: add fun emoji
```
Comment (davidholyko) [inline src/app.js:102]: we can add an emoji and remove this todo

File: src/utils/googleCalendarUtil.js [hunk 7]
```diff
@@ -18,10 +18,25 @@ class GoogleCalendarUtil {
...
   }
 
+  // add check for for if this function was called today
   async createVacationEvent(slackName) {
     await this.createTodayEvent(slackName, 'vacation');
   }
 
+  async getAllSingleDayEvents() {
+    try {
+      let response = await this.calendar.events.list({
```
Comment (davidholyko) [inline src/utils/googleCalendarUtil.js:28]: `const`

File: src/utils/googleCalendarUtil.js [hunk 8]
```diff
@@ -18,10 +18,25 @@ class GoogleCalendarUtil {
...
+        calendarId: GOOGLE_CALENDAR_ID,
+        timeMin: moment().startOf('day').format(),
+        timeMax: moment().endOf('day').format(),
+      });
+      let items = response['data']['items'];
```
Comment (davidholyko) [inline src/utils/googleCalendarUtil.js:33]: `const`

File: src/utils/googleCalendarUtil.js [hunk 9]
```diff
@@ -18,10 +18,25 @@ class GoogleCalendarUtil {
     });
   }
 
+  // add check for for if this function was called today
```
Comment (davidholyko) [inline src/utils/googleCalendarUtil.js:21]: is this a `TODO:` ?

File: src/app.js (hunk 4 above)
Comment (PatrickDohn) [reply src/app.js]: fixed

File: src/app.js (hunk 5 above)
Comment (PatrickDohn) [reply src/app.js]: fixed

File: src/utils/googleCalendarUtil.js (hunk 7 above)
Comment (PatrickDohn) [reply src/utils/googleCalendarUtil.js]: fixed

File: src/utils/googleCalendarUtil.js (hunk 9 above)
Comment (PatrickDohn) [reply src/utils/googleCalendarUtil.js]: Yea that was a note to myself. I removed it

File: src/utils/googleCalendarUtil.js (hunk 8 above)
Comment (PatrickDohn) [reply src/utils/googleCalendarUtil.js]: fixed

File: src/app.js (hunk 6 above)
Comment (PatrickDohn) [reply src/app.js:102]: Do we want one for vacation days as well?

File: src/app.js (hunk 2 above)
Comment (PatrickDohn) [reply src/app.js]: fixed

File: src/utils/googleCalendarUtil.js (hunk 1 above)
Comment (PatrickDohn) [reply src/utils/googleCalendarUtil.js]: fixed

File: src/app.js (hunk 6 above)
Comment (davidholyko) [reply src/app.js:102]: sure

File: src/app.js [hunk 10]
```diff
@@ -88,10 +88,20 @@ const onResponse = async (response) => {
...
-        await GoogleSheetUtil.setTodayAsSickDay(slackName);
-        await GoogleCalendarUtil.createSickEvent(slackName);
-        // TODO: add fun emoji
-        say(`Hello <@${event.user}>! Sick day marked. Hope you feel better!`);
+        const items = await GoogleCalendarUtil.getAllSingleDayEvents();
+        const eventList = [];
+        const summaryCheck = `${slackName}: Sick OOO`;
+        _.map(items, (i) => {
+          eventList.push(i.summary);
+        });
```
Comment (davidholyko) [inline src/app.js:96]: this logic seems like it should belong in the `GoogleCalendarUtil`

**Review by davidholyko — APPROVED** [review]
LGTM

**Pull request merged**